src/ja/common/proxy/command_handler.py

```diff
@@ -17,7 +22,48 @@ class CommandHandler(ABC):
     success property of the Response is True if both validations were passed
     and the Command was executed without error, and is false otherwise.
     """
-    def __init__(self, socket: str):
+    def __init__(self, socket_path: str):
         """!
-        @param socket: The Unix named socket to listen for Commands on.
+        @param socket_path: The Unix named socket to listen for Commands on.
         """
+        self._socket_path = socket_path
+        self._listen_thread = Thread(target=self._listen)
+        self._listen_thread.daemon = True  # Ensures that the thread is killed when main thread exits
+        self._listen_thread.start()
+
+    def _listen(self) -> None:
+        # Make sure the socket does not already exist
+        try:
+            os.unlink(self._socket_path)
+        except OSError:
+            if os.path.exists(self._socket_path):
+                raise
+        named_socket = socket.socket(family=socket.AF_UNIX, type=socket.SOCK_STREAM)
+        named_socket.bind(self._socket_path)
+        named_socket.listen(1)
+
+        while True:
+            connection, client_address = named_socket.accept()
+            try:
+                input_string = ""
+                while True:
+                    data = connection.recv(1024)
+                    if data[-1:] == b"\0":
```

Comment:
A cleaner solution would be to always interpret the first 4 bytes or so as the length of the message, then reading that much bytes.

Reply:
I agree.
I pushed a fixup.